### PR TITLE
RavenDB-11816 - Replace legacy properties when migrating indexes from a v3.x dump

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-11816

### CODE REVIEW

File: src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs [hunk 1]
```diff
@@ -225,6 +231,16 @@ private static IndexDefinition ReadLegacyIndexDefinition(BlittableJsonReaderObje
             return indexDefinition;
         }
 
+        private static string ReplaceLegacyProperties(string str)
+        {
+            if (string.IsNullOrWhiteSpace(str))
+                return str;
+
+            str = str.Replace("new Raven.Abstractions.Linq.DynamicList", string.Empty);
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs]: I think it should be changed to 'DynamicArray' ?
Check here:
https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs#L1464

Comment (ppekrol) [reply src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs]: in v3.x we had: https://github.com/ravendb/ravendb/blob/v3.5/Raven.Client.Lightweight/Indexes/ExpressionStringBuilder.cs#L1588

File: src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs [hunk 2]
```diff
@@ -142,6 +144,8 @@ private static IndexDefinition ReadLegacyIndexDefinition(BlittableJsonReaderObje
 
             if (indexDefinition.Maps != null)
             {
+                indexDefinition.Maps = indexDefinition.Maps.Select(ReplaceLegacyProperties).ToHashSet();
```
Comment (ayende) [inline src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs:147]: Shouldn't use _hash_ set. The user will want to get them in the same order, and if they have duplicates (rare, but possible) we want to maintain them.

Comment (ppekrol) [reply src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs:147]: Talked with @ayende . HashSet is valid at this point. No changes needed here.

File: src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs]: done

File: src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs [hunk 3]
```diff
@@ -225,6 +231,16 @@ private static IndexDefinition ReadLegacyIndexDefinition(BlittableJsonReaderObje
             return indexDefinition;
         }
 
+        private static string ReplaceLegacyProperties(string str)
+        {
+            if (string.IsNullOrWhiteSpace(str))
+                return str;
+
+            str = str.Replace("new Raven.Abstractions.Linq.DynamicList", "DynamicArray");
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/Processors/IndexProcessor.cs]: `DynamicArray` -> `new DynamicArray` ?

**Pull request merged**